test/server/core/abilities/keyword/Smuggle.spec.js

```diff
@@ -0,0 +1,72 @@
+describe('Smuggle keyword', function() {
+    integration(function() {
+        describe('When a card with a Smuggle cost is in resources', function() {
+            beforeEach(function () {
+                this.setupTest({
+                    phase: 'action',
+                    player1: {
+                        groundArena: ['wampa'],
+                        hand: [],
+                        deck: ['mercenary-gunship'],
+                        resources: ['armed-to-the-teeth',
+                            'collections-starhopper',
+                            'smugglers-aid',
+                            'chewbacca#pykesbane',
+                            'battlefield-marine', 'atst', 'atst', 'atst', 'atst', 'atst', 'atst',
+                            'atst', 'atst', 'atst', 'atst', 'atst', 'atst', 'atst'
+                        ], // TODO add a way to make it easier to mix named and generic resources
+                        leader: 'leia-organa#alliance-general',
+                        base: 'administrators-tower'
+                    },
+                    player2: {
+                    }
+                });
+            });
+
+            it('a unit can be played for its smuggle cost', function () {
+                expect(this.player1.countSpendableResources()).toBe(18); // Sanity check before we Smuggle
+                this.player1.clickCard(this.collectionsStarhopper);
+                expect(this.collectionsStarhopper).toBeInLocation('space arena');
+                expect(this.collectionsStarhopper.exhausted).toBe(true);
+                expect(this.player1.countExhaustedResources()).toBe(3);
+                expect(this.player1.countSpendableResources()).toBe(15);
+                expect(this.mercenaryGunship).toBeInLocation('resource');
+            });
+
+            it('an upgrade can be played for its smuggle cost', function () {
+                expect(this.player1.countSpendableResources()).toBe(18); // Sanity check before we Smuggle
+                this.player1.clickCard(this.armedToTheTeeth);
+                expect(this.armedToTheTeeth).toBeInLocation('ground arena');
+                expect(this.wampa.upgrades).toContain(this.armedToTheTeeth);
+
+                // This costs 6 due to the lack of a red aspect on base or leader
+                expect(this.player1.countExhaustedResources()).toBe(6);
+                expect(this.player1.countSpendableResources()).toBe(12);
+                expect(this.mercenaryGunship).toBeInLocation('resource');
+            });
+
+            it('an event can be played for its smuggle cost', function () {
```

Comment:
Would you mind changing this test to be covert strength instead of smuggler's aid? Reason being that it would be nice to do a check at the targeting stage that the swap has already resolved (event in discard, top card in resource line)